server/svix-server/tests/e2e_attempt.rs

```diff
@@ -145,3 +145,191 @@ async fn test_list_attempts_by_endpoint() {
     receiver_1.jh.abort();
     receiver_2.jh.abort();
 }
+
+#[tokio::test]
+async fn test_pagination() {
+    let (client, _jh) = start_svix_server();
+
+    // Setup three endpoints and six messages so there's a sufficient number to test pagination
+    let app = create_test_app(&client, "app1").await.unwrap();
+
+    let mut receivers = Vec::new();
+    for _ in 0..3 {
+        receivers.push(TestReceiver::start(StatusCode::OK));
+    }
+
+    let mut eps = Vec::new();
+    for receiver in &receivers {
+        eps.push(
+            create_test_endpoint(&client, &app.id, &receiver.endpoint)
+                .await
+                .unwrap(),
+        );
+    }
+
+    let mut messages = Vec::new();
+    for i in 1..=6usize {
+        messages.push(
+            create_test_message(
+                &client,
+                &app.id,
+                serde_json::json!({
+                    "test": i,
+                }),
+            )
+            .await
+            .unwrap(),
+        );
+    }
+
+    // Wait until all attempts were made
+    run_with_retries(|| async {
+        for endp_id in eps.iter().map(|ep| &ep.id) {
+            let list: ListResponse<MessageAttemptOut> = client
+                .get(
+                    &format!("api/v1/app/{}/attempt/endpoint/{}/", app.id, endp_id),
+                    StatusCode::OK,
+                )
+                .await
+                .unwrap();
+
+            if list.data.len() != 6 {
+                anyhow::bail!("list len {}, not 6", list.data.len());
+            }
+        }
+
+        Ok(())
+    })
+    .await
+    .unwrap();
+
+    // By endpoint
+    for ep in &eps {
+        let all_attempts: ListResponse<MessageAttemptOut> = client
+            .get(
+                &format!("api/v1/app/{}/attempt/endpoint/{}/", app.id, ep.id),
+                StatusCode::OK,
+            )
+            .await
+            .unwrap();
+
+        // Test Limit
+        let first_three: ListResponse<MessageAttemptOut> = client
+            .get(
+                &format!("api/v1/app/{}/attempt/endpoint/{}/?limit=3", app.id, ep.id),
+                StatusCode::OK,
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(all_attempts.data.len(), 6);
+        assert_eq!(first_three.data.len(), 3);
+
+        assert_eq!(all_attempts.data[0], first_three.data[0]);
+        assert_eq!(all_attempts.data[1], first_three.data[1]);
+        assert_eq!(all_attempts.data[2], first_three.data[2]);
```

Comment:
```suggestion
        assert_eq!(&all_attempts.data[0..=2], &first_three.data);
```
Shouldn't this do the trick?